### DISCUSSION TITLE
Differences between codecs

### DISCUSSION THREAD

**Discussion by Freddiefora** [enzo1982/freac#429] [Audio Format Discussion]
With the useful Freac MEH facility I have undertaken some interesting comparisons of just three codecs, the Monkeys Audio APE format, the latest FLAC format and Apple Lossless, using some very high quality recordings.
As an audio engineer who has recorded many classical concerts and, more recently, jazz concerts as well as concert bands and folk musicians, I understand music and what real instruments sound like. I was interested therefore in how accurately these three formats reproduced the natural timbre of different instruments. This is more complex than it sounds as instruments such as violins, bassoons, oboes, flutes, saxophones, trumpets and, of course the piano, produce multiple harmonics. Capturing these will produce a reasonable facsimilie of the timbre of the instrument.
My findings are as follows; The APE format produces a slightly soft rendering, missing some detail, but it does seem to capture the spatial information quite well. It would be well suited to choral music and opera.
The FLAC format produces good detail but is a little artificial in its rendering of real instruments. They sound clear and bright but not quite as they should be. The sound is a little hard and inflexible, but impressive nonetheless. It is quite well suited to popular music.
The Apple Lossless format was a surprise. Instruments sounded as they should do, with the natural timbre preserved. Listening to Beethoven's Septet for example, it was easy to hear and appreciate the different wind instruments. Spatial information was well preserved and there was a clarity which was there but not over stated. 
So, in my opinion, of the three, Apple Lossless, as implemented in Freac, was superior. I am now comparing some new rips from favourite CDs with the existing versions made with FLAC. Its a pity there is not more hardware support for WAVPACK, TAK and one or two others that Freac supports. Any thoughts?

**Comment by Lee-Carre** [comment]
Unless there are genuine bugs (but, then, FLAC self-validation would (very likely) fail), or hardware defects, then lossless codecs are indistinguishable, because the decoded waveform matches the original input waveform.

Do you have, for example, spectral analysis to support your assertions?

If you've found genuine bugs, then I'm sure the Xiph developers would be keen to hear.

What steps have you taken to eliminate bias and other psych/perceptual effects in comparisons, from unduly influencing your conclusions?

> Its a pity there is not more hardware support for WAVPACK, TAK and one or two others that Freac supports. Any thoughts?

Use uncompressed files, in that case. Avoids all the problems you describe.

---

All this vague new-age-sounding talk about the ‘colour’ of the sound very much reminds me of the apt remarks which Steve Albini made in his *The Problem With Music* essay. I propose a drinking-game; every time the word ‘warm’ is mentioned as a description of audio (even in the negative; eg ‘not warm’ or ‘cold’), drink another shot.
